native/src/main/java/io/ballerina/stdlib/sql/nativeimpl/QueryProcessor.java

```diff
@@ -66,61 +74,113 @@ private QueryProcessor() {
             BObject client, Object paramSQLString,
             Object recordType,
+            AbstractStatementParameterProcessor statementParameterProcessor,
+                AbstractResultParameterProcessor resultParameterProcessor) {
+        ResultSet resultSet = null;
+        try {
+            resultSet = initQuery(client, paramSQLString, statementParameterProcessor);
+            RecordType streamConstraint = (RecordType) ((BTypedesc) recordType).getDescribingType();
+            List<ColumnDefinition> columnDefinitions = Utils.getColumnDefinitions(resultSet, streamConstraint);
+            return ValueCreator.createStreamValue(TypeCreator.createStreamType(streamConstraint,
+                    PredefinedTypes.TYPE_NULL), resultParameterProcessor
+                    .createRecordIterator(resultSet, statement, connection, columnDefinitions, streamConstraint));
+        } catch (SQLException e) {
+            Utils.closeResources(trxResourceManager, resultSet, statement, connection);
+            BError errorValue = ErrorGenerator.getSQLDatabaseError(e,
+                    "Error while executing SQL query: " + sqlQuery + ". ");
+            return ValueCreator.createStreamValue(TypeCreator.createStreamType(Utils.getDefaultStreamConstraint(),
+                    PredefinedTypes.TYPE_NULL), createRecordIterator(errorValue));
+        } catch (ApplicationError applicationError) {
+            Utils.closeResources(trxResourceManager, resultSet, statement, connection);
+            BError errorValue = ErrorGenerator.getSQLApplicationError(applicationError.getMessage());
+            return getErrorStream(recordType, errorValue);
+        } catch (Throwable e) {
+            Utils.closeResources(trxResourceManager, resultSet, statement, connection);
+            String message = e.getMessage();
+            if (message == null) {
+                message = e.getClass().getName();
+            }
+            BError errorValue = ErrorGenerator.getSQLApplicationError(
+                    "Error while executing SQL query: " + sqlQuery + ". " + message);
+            return getErrorStream(recordType, errorValue);
+        }
+    }
+
+    public static Object nativeQueryRow(
+            BObject client, Object paramSQLString,
+            BTypedesc ballerinaType,
             AbstractStatementParameterProcessor statementParameterProcessor,
             AbstractResultParameterProcessor resultParameterProcessor) {
+        ResultSet resultSet = null;
+        try {
+            resultSet = initQuery(client, paramSQLString, statementParameterProcessor);
+
+            if (!resultSet.next()) {
+                throw new ApplicationError("Query retrieved an empty result.");
+            }
+
+            Type describingType = ballerinaType.getDescribingType();
+            Object returnValue;
+
+            // If the return data type is a record
+            if (describingType.getTag() == TypeTags.RECORD_TYPE_TAG) {
```

Comment:
1. Can the user give `record{}|int`? 
2. Have we handled typed records, i.e `Student|sql:Error value = client->queryRow(.....)`

Reply:
1. Did not consider this use case. Will need to discuss how the handle this.
2. It should work correctly. Will add test cases to verify.

Reply:
Test cases for (2) have been added in eaf6cfc

Reply:
For the first use case we have to throw an error I think. We only expect the user to give one type. Could you check whether the compiler allows passing union type description which has more than one type? Note this is after the error type is removed.

Reply:
Fixed in b9bd369